### ISSUE TITLE
CMakeLists.txt: find_package(dlib) fails on Fedora/Red Hat

### ISSUE THREAD

**Issue by kimwalisch** [davisking/dlib#1137]
Hi,

When installing ```dlib-devel``` using the package manager of Fedora or Red Hat ```find_package(dlib)``` fails when I try to use dlib in my ```CMakeLists.txt```:

```CMake
cmake_minimum_required(VERSION 3.1)
find_package(dlib REQUIRED)

add_executable(example example.cpp)
target_link_libraries(example dlib::dlib_shared)
```

Here is the error message:

```
cmake .
CMake Error at /usr/local/lib/cmake/dlib/dlib.cmake:85 (message):
  The imported target "dlib::dlib" references the file

     "/usr/local/lib/libdlib.a"

  but this file does not exist.  Possible reasons include...
```

The error is caused by a Fedora/Red Hat policy that static libraries must be put into its own package (e.g. ```dlib-static```), hence ```dlib-devel``` contains only the shared dlib library. The ```dlib-devel``` package also installs the CMake config files which export the ```dlib::dlib``` target for the static dlib library which is missing in the ```dlib-devel``` package. Because of this ```find_package(dlib)``` fails. 

In the ```dlib.spec``` file which is used to build the official dlib RPM packages the static dlib library is simply deleted after the build has finished:

```spec
%install
pushd build
%make_install
popd
rm -f %{buildroot}/%{_libdir}/*.a
```

There is no easy workaround I know that could be used in the ```dlib.spec``` to fix the official dlib RPM packages (The only workaround I know is to include dlib.a in the ```dlib-devel``` package which would fix the issue but this is not allowed by Fedora/Red Hat).

**Comment by davisking** [comment]
I wasn't involved or even knew about the creation of the dlib RPM package.
The package maintainers need to test their packages before they publish
them to make sure they haven't somehow broken the package. I don't know
what you want me to do about this. You should complain to the package
maintainer. Maybe there is some change to the dlib CMake files that might
make the package maintainer's life easier, but we have to begin from a
point where the maintainer is at least testing their packages before
pushing them out.

**Comment by davisking** [comment]
I also want to reference this discussion: https://github.com/davisking/dlib/issues/923.  The main issue is that the vast majority of dlib users have no idea what a shared library is and if the build system produces a shared library they won't know what to do, will get confused, and then post questions in the dlib forums or email me. This seems crazy to experienced C++ users and seemed crazy (really still does seem crazy) to me until I ran an open source project for years. Because of this issue I've configured things so that it's intentionally difficult to use dlib as a shared library. I want people to link to it statically unless they really understand what they are doing.
